python/nano/tutorial/notebook/inference/pytorch/pytorch_context_manager.ipynb

```diff
@@ -0,0 +1,322 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[View the runnable example on GitHub](https://github.com/intel-analytics/BigDL/tree/main/python/nano/tutorial/notebook/inference/pytorch/pytorch_context_manager.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Automatic inference context management by `get_context`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can use ``InferenceOptimizer.get_context(model=...)`` API to enable automatic context management for PyTorch inference. Only one line of code change, then Nano will automatically provide suitable context management for each accelerated model, it usually contains part of or all of following three types of context manager:\n",
+    "\n",
+    "1. torch.no_grad() to disable gradients, which will be used for all model\n",
+    "   \n",
+    "2. torch.cpu.amp.autocast(dtype=torch.bfloat16) to run in mixed precision, which will be provided for bf16 related model\n",
+    "   \n",
+    "3. torch.set_num_threads() to control thread number, which will be used only if you specify thread_num when `trace/quantize/optimize`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To inference using Bigdl-nano InferenceOptimizer, the following packages need to be installed first. We recommend you to use [Miniconda](https://docs.conda.io/en/latest/miniconda.html) to prepare the environment and install the following packages in a conda environment.\n",
+    "\n",
+    "You can create a conda environment by executing:\n",
+    "\n",
+    "```\n",
+    "# \"nano\" is conda environment name, you can use any name you like.\n",
+    "conda create -n nano python=3.7 setuptools=58.0.4\n",
+    "conda activate nano\n",
+    "```\n",
+    "> 📝 **Note**\n",
+    ">\n",
+    "> During your installation, there may be some warnings or errors about version, just ignore them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Necessary packages for inference accelaration\n",
+    "!pip install --pre --upgrade bigdl-nano[pytorch,inference]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we take a pretrained ResNet18 model for example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from torchvision.models import resnet18\n",
+    "\n",
+    "model = resnet18(pretrained=True)\n",
+    "model.eval()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. `InferenceOptimizer.trace`\n",
```

Comment:
I would recommended to use section headers for the "1. `InferenceOptimizer.trace`", "2. `InferenceOptimizer.quantize`" and "3. `InferenceOptimizer.optimize`" instead of using lists :)  It could make the section more clear for readers to find the contents they want.